test/test_vmap.py

```diff
@@ -3490,6 +3487,45 @@ def test_randperm(self, device, randomness, use_generator):
             for i in range(B0):
                 assert torch.allclose(vmap_result[i], expected)
 
+    @parametrize('use_generator', [True, False])
+    @parametrize('randomness', ['error', 'same', 'different'])
+    def test_random_inplace_not_batched(self, device, use_generator, randomness):
+        # tests that when in place random ops are being called during vmap but not with a batched tensor,
+        # it behaves like a normal random_ when using same random and errors during different random
+        supported_ops = [
+            lambda t, _: t.random_(**kwargs),
+            lambda t, _: t.random_(100, **kwargs),
+            lambda t, _: t.random_(-5, 100, **kwargs),
+            lambda t, _: t.normal_(**kwargs),
+        ]
+
+        B0 = 4
+        seed = 1234567
+        generator = torch.Generator(device=device)
+        orig_state = generator.get_state()
+        kwargs = {'generator': generator} if use_generator else {}
+
+        for op in supported_ops:
+            vmaped_value, unvmaped_value = torch.randn(B0, B0, device=device), torch.randn(B0, B0, device=device)
```

Comment:
Why is it `torch.randn(B0, B0)`? "B0" is generally used in this file as the name for a dimension that gets vmapped over (i.e. a batch dimension). It would make the test a bit clearer if we did something like:

vmaped_value, unvmaped_value = torch.randn(B0, device=device), torch.randn(3, 4, device=device)